File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0;
        unordered_map<char,int>us;
        int lo=0,hi=0;
        while(hi<s.size()){
            if(us.find(s[hi])!=us.end()){
                lo = max(lo,us[s[hi]]+1);
            }
            us[s[hi]]=hi;
            ans = max(ans,hi-lo+1);
            ++hi;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/insert-new-interval/submission-5.py
class Solution:
    def insert(self, inter: List[List[int]], ninter: List[int]) -> List[List[int]]:

        ans, idx = [], 0
        while idx<len(inter) and inter[idx][1]< ninter[0]:
            ans.append(inter[idx])
            idx +=1 

        while idx<len(inter) and inter[idx][0]<=ninter[1]:
            ninter[0] = min(ninter[0], inter[idx][0])
            ninter[1] = max(ninter[1], inter[idx][1])
            idx +=1 

        ans.append(ninter)

        while idx <len(inter):
            ans.append(inter[idx])
            idx+=1
            
        return ans


        